services/chronikService.ts

```diff
@@ -387,22 +387,58 @@ export class ChronikBlockchainClient {
     }
   }
 
-  private getSortedInputAddresses (transaction: Tx): string[] {
+  private getSortedInputAddresses (transaction: Tx): Array<{address: string, amount: Prisma.Decimal}> {
     const addressSatsMap = new Map<string, bigint>()
-
     transaction.inputs.forEach((inp) => {
       const address = outputScriptToAddress(this.networkSlug, inp.outputScript)
       if (address !== undefined && address !== '') {
         const currentValue = addressSatsMap.get(address) ?? 0n
         addressSatsMap.set(address, currentValue + inp.sats)
       }
     })
-
     const sortedInputAddresses = Array.from(addressSatsMap.entries())
       .sort(([, valueA], [, valueB]) => Number(valueB - valueA))
-      .map(([address]) => address)
+    return sortedInputAddresses.map(([address, sats]) => {
+      const format = xecaddr.detectAddressFormat(address)
```

Comment:
This is unnecessary. We already know the network, accessible from `this.networkSlug`. 
Best to define something like `decimalCases` outside the `.map` to be either `1e2` or `1e8` based on that, and then here, inside the `.map`, only apply it, to avoid this redundant checks that could slow the code overall if they are ran many times in parallel (like when a block arrives)